libs/ai-endpoints/tests/unit_tests/test_stop.py

```diff
@@ -0,0 +1,124 @@
+                {
+                    "index": 0,
+                    "message": {"role": "assistant", "content": "Ok"},
+                }
+            ],
+        },
+    )
+
+
+def test_stop_property_invoke(requests_mock: Mocker) -> None:
+    expected_stop = ["STOP"]
+
+    client = ChatNVIDIA(model="mock-model", api_key="mocked", stop=expected_stop)
+    response = client.invoke("Ok?")
+
+    assert response.content == "Ok"
+
+    assert requests_mock.last_request is not None
+    request_payload = requests_mock.last_request.json()
+    assert "stop" in request_payload
+    assert request_payload["stop"] == expected_stop
+
+
+def test_stop_parameter_invoke(requests_mock: Mocker) -> None:
+    expected_stop = ["STOP"]
+
+    client = ChatNVIDIA(model="mock-model", api_key="mocked")
+    response = client.invoke("Ok?", stop=expected_stop)
+
+    assert response.content == "Ok"
+
+    assert requests_mock.last_request is not None
+    request_payload = requests_mock.last_request.json()
+    assert "stop" in request_payload
+    assert request_payload["stop"] == expected_stop
+
+
+def test_stop_override_invoke(requests_mock: Mocker) -> None:
+    expected_stop = ["STOP"]
+
+    client = ChatNVIDIA(model="mock-model", api_key="mocked", stop=["GO"])
+    response = client.invoke("Ok?", stop=expected_stop)
+
+    assert response.content == "Ok"
+
+    assert requests_mock.last_request is not None
+    request_payload = requests_mock.last_request.json()
+    assert "stop" in request_payload
+    assert request_payload["stop"] == expected_stop
+
+
+def test_stop_property_stream(requests_mock: Mocker) -> None:
+    expected_stop = ["STOP"]
+
+    client = ChatNVIDIA(model="mock-model", api_key="mocked", stop=expected_stop)
+    response = client.stream("Ok?")
+
+    assert next(response).content == "Ok"
+
+    assert requests_mock.last_request is not None
+    request_payload = requests_mock.last_request.json()
+    assert "stop" in request_payload
+    assert request_payload["stop"] == expected_stop
+
+
+def test_stop_parameter_stream(requests_mock: Mocker) -> None:
+    expected_stop = ["STOP"]
+
+    client = ChatNVIDIA(model="mock-model", api_key="mocked")
+    response = client.stream("Ok?", stop=expected_stop)
+
+    assert next(response).content == "Ok"
+
+    assert requests_mock.last_request is not None
+    request_payload = requests_mock.last_request.json()
+    assert "stop" in request_payload
+    assert request_payload["stop"] == expected_stop
+
+
+def test_stop_override_stream(requests_mock: Mocker) -> None:
+    expected_stop = ["STOP"]
+
+    client = ChatNVIDIA(model="mock-model", api_key="mocked", stop=["GO"])
+    response = client.stream("Ok?", stop=expected_stop)
+
+    assert next(response).content == "Ok"
+
+    assert requests_mock.last_request is not None
+    request_payload = requests_mock.last_request.json()
+    assert "stop" in request_payload
+    assert request_payload["stop"] == expected_stop
```

Comment:
I have a suggestion for the repeating test cases with invoke/stream, these can be combined with [@pytest.mark.parametrize](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html) and stop can be passed as parameter.

Reply:
will you give me an example?

Reply:
Sure,

this 2 can be combined as

def test_stop_property_invoke(requests_mock: Mocker) -> None:
    expected_stop = ["STOP"]

    client = ChatNVIDIA(model="mock-model", api_key="mocked", stop=expected_stop)
    response = client.invoke("Ok?")

    assert response.content == "Ok"

    assert requests_mock.last_request is not None
    request_payload = requests_mock.last_request.json()
    assert "stop" in request_payload
    assert request_payload["stop"] == expected_stop


def test_stop_parameter_invoke(requests_mock: Mocker) -> None:
    expected_stop = ["STOP"]

    client = ChatNVIDIA(model="mock-model", api_key="mocked")
    response = client.invoke("Ok?", stop=expected_stop)

    assert response.content == "Ok"

    assert requests_mock.last_request is not None
    request_payload = requests_mock.last_request.json()
    assert "stop" in request_payload
    assert request_payload["stop"] == expected_stop

======================================================================
@pytest.mark.parametrize("client_stop, expected_stop", [(None, ["STOP"]), (["GO"], ["STOP"])])
def test_stop_invoke(requests_mock: Mocker, client_stop, expected_stop) -> None:
    client = ChatNVIDIA(model="mock-model", api_key="mocked", stop=client_stop)
    response = client.invoke("Ok?", stop=expected_stop)

    assert response.content == "Ok"

    assert requests_mock.last_request is not None
    request_payload = requests_mock.last_request.json()
    assert "stop" in request_payload
    assert request_payload["stop"] == expected_stop

Reply:
nice idea. i'll update to use this.

Reply:
ptal